google-ads-examples/src/main/java/com/google/ads/googleads/examples/reporting/ParallelReportDownload.java

```diff
@@ -0,0 +1,249 @@
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    List<Long> customerIds;
+
+    @Parameter(names = ArgumentNames.LOGIN_CUSTOMER_ID)
+    Long loginCustomerId;
+  }
+
+  public static void main(String[] args) {
+    ParallelReportDownloadParams params = new ParallelReportDownloadParams();
+    if (!params.parseArguments(args)) {
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerIds = ImmutableList.of(Long.valueOf("INSERT CUSTOMER IDS"));
+      // Optionally specify the login customer ID if your access to the CIDs is via a manager
+      // account.
+      // params.loginCustomerId = Long.parseLong("INSERT_LOGIN_CUSTOMER_ID");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      GoogleAdsClient.Builder builder = GoogleAdsClient.newBuilder().fromPropertiesFile();
+      if (params.loginCustomerId != null) {
+        builder.setLoginCustomerId(params.loginCustomerId);
+      }
+      googleAdsClient = builder.build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new ParallelReportDownload().runExample(googleAdsClient, params.customerIds);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example. This method may block for up to MAX_WAIT_TIME_SECONDS while waiting for
+   * reports to complete.
+   *
+   * @param googleAdsClient the client library instance for API access.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, List<Long> customerIds) {
+    // Creates a single client which can be shared by all threads.
+    // gRPC handles multiplexing parallel requests to the underlying API connection.
+    try (GoogleAdsServiceClient serviceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      // IMPORTANT: You should avoid hitting the same customer ID in parallel. There are rate limit
+      // errors at the customer ID level, but not at the developer token level. Rate limits are
```

Comment:
>  but not at the developer token level

I'm not sure this will give impression that there is no rate limit at all for dev token level (which is not true)?

Reply:
Updated wording here, PTAL.